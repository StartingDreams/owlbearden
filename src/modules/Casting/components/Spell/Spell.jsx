import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardText } from 'material-ui/Card';

export class Spell extends React.PureComponent {
  render() {
    const { spell } = this.props;
    console.log(spell);
    const subtitle = `level: ${spell.get('level')}, range: ${spell.get('range')}, casting time: ${spell.get('casting_time')}`;
    return (
      <Card>
        <CardHeader
          title={spell.get('name')}
          subtitle={subtitle}
          actAsExpander
          showExpandableButton
        />
        <CardText expandable>
          <ul>
            <li>Level: {spell.get('level')}</li>
            <li>Range: {spell.get('range')}</li>
            <li>Ritual: {spell.get('ritual')}</li>
            <li>School: {spell.get('school')}</li>
            <li>Casting Time: {spell.get('casting_time')}</li>
            <li>Components: {spell.get('components')}</li>
            <li>Duration: {spell.get('duration')}</li>
            <li>Concentration: {spell.get('concentration')}</li>
            <li>Cast at higher level: {spell.get('higher_level')}</li>
          </ul>
          <p>{spell.get('desc').join('<br/>')} {spell.get('page')}</p>
        </CardText>
      </Card>
    );
  }
}

Spell.propTypes = {
  spell: ImmutablePropTypes.contains({
    name: PropTypes.string.isRequired,
  }),
};

export default Spell;
