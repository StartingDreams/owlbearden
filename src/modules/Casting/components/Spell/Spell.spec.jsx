import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import { Spell } from './Spell';

const testSpell = {
  casting_time: '1 action',
  classes: [{
    name: 'Wizard',
    url: 'http://www.dnd5eapi.co/api/classes/12',
  }],
  components: ['V', 'S', 'M'],
  concentration: 'no',
  desc: ['A shimmering green arrow streaks toward a target within range and bursts in a spray of acid. Make a ranged spell attack against the target. On a hit, the target takes 4d4 acid damage immediately and 2d4 acid damage at the end of its next turn. On a miss, the arrow splashes the target with acid for half as much of the initial damage and no damage at the end of its next turn.'],
  duration: 'Instantaneous',
  higher_level: ['When you cast this spell using a spell slot of 3rd level or higher, the damage (both initial and later) increases by 1d4 for each slot level above 2nd.'],
  id: '58f40b80c9e7ce9f72153112',
  index: 1,
  level: 2,
  material: 'Powdered rhubarb leaf and an adderâ€™s stomach.',
  name: 'Acid Arrow',
  page: 'phb 259',
  range: '90 feet',
  ritual: 'no',
  school: {
    name: 'Evocation',
    url: 'http://www.dnd5eapi.co/api/magic-schools/5',
  },
  subclasses: [{
    name: 'Lore',
    url: 'http://www.dnd5eapi.co/api/subclasses/2',
  }, {
    name: 'Land',
    url: 'http://www.dnd5eapi.co/api/subclasses/4',
  }],
  url: 'http://www.dnd5eapi.co/api/spells/1',
};

describe('<Spell />', () => {
  it('spell should match the snapshot', () => {
    const spell = fromJS(testSpell);
    const wrapper = shallow(<Spell spell={spell} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
