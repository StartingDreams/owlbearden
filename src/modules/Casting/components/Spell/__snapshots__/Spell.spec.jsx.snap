// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spell /> spell should match the snapshot 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    actAsExpander={true}
    avatar={null}
    showExpandableButton={true}
    subtitle="Level: 2, Range: 90 feet, Casting Time: 1 action"
    title="Acid Arrow"
  />
  <CardText
    expandable={true}
  >
    <ul>
      <li>
        Level: 
        2
      </li>
      <li>
        Range: 
        90 feet
      </li>
      <li>
        Ritual: 
        no
      </li>
      <li>
        School: 
        Evocation
      </li>
      <li>
        Casting Time: 
        1 action
      </li>
      <li>
        Components: 
        V
        S
        M
      </li>
      <li>
        Duration: 
        Instantaneous
      </li>
      <li>
        Concentration: 
        no
      </li>
      <li>
        Cast at higher level: 
        When you cast this spell using a spell slot of 3rd level or higher, the damage (both initial and later) increases by 1d4 for each slot level above 2nd.
      </li>
    </ul>
    <p>
      A shimmering green arrow streaks toward a target within range and bursts in a spray of acid. Make a ranged spell attack against the target. On a hit, the target takes 4d4 acid damage immediately and 2d4 acid damage at the end of its next turn. On a miss, the arrow splashes the target with acid for half as much of the initial damage and no damage at the end of its next turn. phb 259
    </p>
  </CardText>
</Card>
`;
